### ISSUE TITLE
(aws-stepfunctions): State machine does not allow token in its name

### ISSUE THREAD

**Issue by supergillis** [aws/aws-cdk#13912]
It's not possible to have a token in a state machine name after upgrading from CDK v1.91.0 to v1.95.1.

### Reproduction Steps

```typescript
new stepfunctions.StateMachine(this, 'StateMachine', {
  stateMachineName: `${cdk.Aws.STACK_NAME}-Machine`,
});
```

### What did you expect to happen?

The state machine should be created with the name of the stack plus `-Machine`.

### What actually happened?

The following error occured:

```
State Machine name must match "^[0-9a-zA-Z+!@._-]+$". Received: ${Token[AWS.StackName.9]}-Machine
```

### Environment

  - **CDK CLI Version:** 1.95.1
  - **Framework Version:**  1.95.1
  - **Node.js Version:** v14.16.0
  - **OS:** OS X
  - **Language (Version):** TypeScript

### Other

Introduced in PR #13387.

Validate that the state machine name is not a token and then perform the validation.

https://github.com/aws/aws-cdk/blob/ed2bbe6767f83de6becd9773cb7591c24cc753e3/packages/%40aws-cdk/aws-stepfunctions/lib/state-machine.ts#L434

The following workaround can be used temporarily.

```typescript
const machine = new stepfunctions.StateMachine(this, 'StateMachine', {
    // stateMachineName: `${cdk.Aws.STACK_NAME}-Machine`,
});
(this.resource.node.defaultChild as cdk.CfnResource).addPropertyOverride('StateMachineName', `${cdk.Aws.STACK_NAME}-Machine`);
```

--- 

This is :bug: Bug Report

**Comment by jstrese** [comment]
@shivlaks I think this issue can be closed - this is a duplicate of issue #13946 and resolved in PR #14064.

**Comment by shivlaks** [comment]
@jstrese I think you meant that it's fixed by #13970 
thanks again for contributing the fix. resolving this issue.
